### DISCUSSION TITLE
Python, eglot, LSP et ruff

### DISCUSSION THREAD

**Discussion by raylemon** [abougouffa/minemacs#111] [Q&A] [answered]
Bonjour,
Comment puis-je attacher Ruff-LSP (formatter de code python) avec eglot ? Est-il possible d’utiliser Pyright en même temps ? 

Merci !

**Comment by abougouffa** [comment, marked as answer]
Bonjour @raylemon ,

Pour activer `ruff-lsp`, il suffit de l'ajouter à `eglot-server-programs`. Tu peux utiliser un truc comme ça pour ne pas écraser les autres serveurs LSP de python: 

```elisp
(with-eval-after-load 'eglot
  (setcdr (assoc '(python-mode python-ts-mode) eglot-server-programs)
          (eglot-alternatives
           '("pylsp" "pyls" ("pyright-langserver" "--stdio") "jedi-language-server" "ruff-lsp"))))
```

`M-x eglot` et tu dois avoir le choix de lancer `ruff-lsp`.

En revanche, je ne sais pas si Eglot (ou LSP-mode) supporte deux serveurs en même temps.
